plugins/nodes/src/js/data/MesosMastersHealth.js

```diff
@@ -0,0 +1,52 @@
+import Config from "#SRC/js/config/Config";
+import UnitHealthUtil from "#SRC/js/utils/UnitHealthUtil";
+
+import { Observable } from "rxjs";
+import { request } from "@dcos/http-service";
+import "rxjs/add/observable/of";
+import "rxjs/add/observable/timer";
+
+import "rxjs/add/operator/catch";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/exhaustMap";
+import "rxjs/add/operator/publishReplay";
+
+function fetchUnit(unitID) {
+  const unitUrl = `${Config.rootUrl}${
+    Config.unitHealthAPIPrefix
+  }/units/${unitID}/nodes`;
+
+  return request(unitUrl)
+    .map(response => response.nodes)
+    .catch(_err => Observable.of([]));
+}
+
+function pollStream(interval, stream) {
+  return Observable.timer(0, interval).exhaustMap(() => stream);
+}
+
+export function replayStream(stream) {
+  return stream.publishReplay(1).refCount();
+}
+
+function withHealthDescription(masters) {
+  return masters.map(master =>
+    Object.assign({}, master, {
+      healthDescription: UnitHealthUtil.getHealth(parseInt(master.health, 10))
+    })
+  );
+}
+
+export function mesosMastersHealthQuery(source, interval) {
+  const request$ = source().map(withHealthDescription);
+
+  return replayStream(pollStream(interval, request$));
```

Comment:
How would you like having `request$.pollStream(interval).replayStream()`? It should be possible with `lift`, though a bit of overhead could be involved.

Reply:
Which library will provide the `lift` function or do you propose another way to achieve that interface? I am happy enough that the functions compose (although I do not use the pipe or compose explicitly). Also, is `lift` maybe a confusing concept for the team?

I would love to try it but ATM I don't see how.

Reply:
Lift should be a part of RxJs. Could be an overhead in this particular case, though we could employ `lift` for our custom Observables.

Reply:
I looked on rxjs. I guess they have it internally but I was not able to find that on the docs. I found a pr from 2015 talking about that.

Reply:
http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-lift